airflow/providers/google/suite/hooks/calendar.py

```diff
@@ -0,0 +1,239 @@
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account.
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    def __init__(
+        self,
+        api_version: str,
+        gcp_conn_id: str = 'google_cloud_default',
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+    ) -> None:
+        super().__init__(
+            gcp_conn_id=gcp_conn_id,
+            delegate_to=delegate_to,
+            impersonation_chain=impersonation_chain,
+        )
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self.delegate_to = delegate_to
+        self._conn = None
+
+    def get_conn(self) -> Any:
+        """
+        Retrieves connection to Google Calendar.
+
+        :return: Google Calendar services object.
+        :rtype: Any
+        """
+        if not self._conn:
+            http_authorized = self._authorize()
+            self._conn = build('calendar', self.api_version, http=http_authorized, cache_discovery=False)
+
+        return self._conn
+
+    def get_events(
+        self,
+        calendar_id: str = 'primary',
+        i_cal_uid: Optional[str] = None,
+        max_attendees: Optional[int] = None,
+        max_results: Optional[int] = None,
+        order_by: Optional[str] = None,
+        private_extended_property: Optional[str] = None,
+        q: Optional[str] = None,
+        shared_extended_property: Optional[str] = None,
+        show_deleted: Optional[bool] = False,
+        show_hidden_invitation: Optional[bool] = False,
+        single_events: Optional[bool] = False,
+        sync_token: Optional[str] = None,
+        time_max: Optional[datetime] = None,
+        time_min: Optional[datetime] = None,
+        time_zone: Optional[str] = None,
+        updated_min: Optional[datetime] = None,
+    ) -> list:
+        """
+        Gets events from Google Calendar from a single calendar_id
+        https://developers.google.com/calendar/api/v3/reference/events/list
+
+        :param calendar_id: The Google Calendar ID to interact with
+        :type calendar_id: str
+        :param i_cal_uid: Optional. Specifies event ID in the iCalendar format in the response.
```

Comment:
What is iCalendar?

Reply:
Quite popular calendar media format ([wiki](https://en.wikipedia.org/wiki/ICalendar))

Reply:
Good to know!